src/modules/deal/deal.service.ts

```diff
@@ -236,4 +238,16 @@ export class DealService {
       },
     };
   }
+
+  private async updateBorrowingRestrictionForNewDeal(dealIdentifier: string): Promise<void> {
+    try {
+      await this.dealBorrowingRestrictionService.updateBorrowingRestrictionForDeal(dealIdentifier);
+    } catch (error: unknown) {
+      // TODO APIM-170: do we want more logging here?
+      throw new InternalServerErrorException('Internal server error', {
+        cause: error as Error,
+        description: `Failed to update the deal borrowing restriction after creating deal ${dealIdentifier}`,
+      });
```

Comment:
I'd be interested to know what you think here @avaitonis. We'll already get the usual logs that an endpoint has errored and it'll include the error object. Do you think we need an extra, separate log message to highlight the unusual state here (where a deal has been created but the borrowing restriction hasn't been updated) or are you happy without that?

If you're not sure then I'll leave it as-is for now and we can easily add it later if we find that it'd be useful

Reply:
I think it is fine for now.
I still don't know how Azure logs will look, how easy is to find exceptions, responses with 500 and other error codes.
I would prefer to get notification email with enough error details to understand what happened. If notification email is vague, then some issues can be missed. @IgnatG @abhi-markan 

Recently we had exactly this error "Failed to update the deal borrowing restriction after creating deal" because of timeout and even knowing about it didn't help to solve it.